packages/orchestration/src/orchestration.js

```diff
@@ -179,18 +179,42 @@ export const OrchestrationI = M.interface('Orchestration', {
   createAccount: M.callWhen(M.string(), M.string()).returns(
     M.remotable('ChainAccount'),
   ),
+  createQueryConnection: M.callWhen(M.string()).returns(
+    M.remotable('Connection'),
+  ),
 });
 
+/** @typedef {{ icqControllerNonce: number; icaControllerNonce: number; }} PortNonceCounters */
+/** @typedef {{ powers: PowerStore; queryPort: Port | undefined; } & PortNonceCounters } OrchestrationState */
+
+// XXX temporary until #9165 (vat-network tracks nonces)
+/** @type {Record<string, {portType: string; nonceKey: keyof PortNonceCounters}>} */
+const portOpts = {
+  ica: {
+    portType: 'icacontroller',
+    nonceKey: 'icaControllerNonce',
+  },
+  icq: {
+    portType: 'icqcontroller',
+    nonceKey: 'icqControllerNonce',
+  },
+};
+
 /**
  * @param {Zone} zone
- * @param {ReturnType<typeof prepareChainAccount>} createChainAccount
+ * @param {ReturnType<prepareChainAccount>} createChainAccount
+ * @param {ReturnType<prepareQueryConnection>} createQueryConnection
  */
-const prepareOrchestration = (zone, createChainAccount) =>
+const prepareOrchestration = (
+  zone,
+  createChainAccount,
+  createQueryConnection,
+) =>
   zone.exoClassKit(
     'Orchestration',
     {
       self: M.interface('OrchestrationSelf', {
-        bindPort: M.callWhen().returns(M.remotable()),
+        bindPort: M.callWhen(M.record()).returns(M.remotable()),
```

Comment:
if you meant that it can take a promise for a record, that needs `M.await()`.
That would be unusual, though.